src/Symfony/Component/Validator/Constraints/ChoiceValidator.php

```diff
@@ -59,7 +59,7 @@ public function validate($value, Constraint $constraint)
         }
 
         if (false === $constraint->strict) {
-            @trigger_error('Setting the strict option of the Choice constraint to false is deprecated since version 3.2 and will be removed in 4.0.', E_USER_DEPRECATED);
+            throw new \RuntimeException('The "strict" option of the Choice constraint should not be used.');
```

Comment:
We cannot completely remove the option as that would break code that was valid in 3.4 (as the base `Constraint` class would complain about an unknown option while setting the `strict` option to `true` is the forward compatible change to be made in 3.x).